### ISSUE TITLE
Moonlight crashes when the connection stutters

### ISSUE THREAD

**Issue by richardios275** [moonlight-stream/moonlight-qt#891]
**Describe the bug**
As stated in the title, the program just quits when the connection stutters. It happens so often that it bugs me. I've only tested this in Windows

**Steps to reproduce**
Cutting off Wi-Fi for momentarily doesn't seem to reproduce the bug, I think it's best to try see if it happens to you by using Wi-Fi to connect, my 'Server PC' is also connected via Wi-Fi

**Screenshots**
If applicable, add screenshots to help explain your problem. If the issue is related to video glitching or poor quality, please include screenshots.

**Other Moonlight clients**
- Does the issue occur when using Moonlight on iOS or Android?

**Moonlight settings (please complete the following information)**
I had Frame Pacing on and Hardware acceleration to forced, which I think are the most relevant settings on this bug, but it appears to 

**Client PC details (please complete the following information)**
 - OS: Windows 11 21H2
 - Moonlight Version: 4.3.0
 - GPU: Intel UHD 600

**Server PC details (please complete the following information)**
 - OS: Windows 11 21H2
 - GeForce Experience version: 3.26.0.154
 - Nvidia GPU driver: 522.25
 - Antivirus and firewall software: Just Windows Defender

**Moonlight Logs
[Moonlight-1667320291.log](https://github.com/moonlight-stream/moonlight-qt/files/9912335/Moonlight-1667320291.log)
e attach)**

**Additional context**
Could just be a coincidence, but I did end up with an RTSP connection error after one of the crashes which leads me to restarting the server PC

**Comment by cgutman** [comment]
Can you upload the Moonlight.dmp file? It should be in the temp folder next to the log files.

**Comment by cgutman** [comment]
Please also try the latest nightly build which has a fix for a bug that could cause this issue: https://ci.appveyor.com/project/cgutman/moonlight-qt/builds/45250840/job/je44yxvsnahgkksl/artifacts

**Comment by transient-sepia** [comment]
Hello, I experience something eerily familiar, but with a Linux client. I didn't have a problem like this on previous versions of Moonlight.

**Client PC details (please complete the following information)**

    OS: Arch Linux
    Moonlight Version: 4.3.0
    GPU: Intel UHD 620

**Server PC details (please complete the following information)**

    OS: Windows 10 22H2
    GeForce Experience version: 3.26.0.154
    Nvidia GPU driver: 526.47
    Antivirus and firewall software: Just COMODO Firewall

**Moonlight Logs (please attach)**
[Moonlight-20221102.log](https://github.com/moonlight-stream/moonlight-qt/files/9923879/Moonlight-20221102.log)

**Additional context**

Just before a coredump there are messages like these in system journal:

```
Nov 02 20:57:09 nb-t480s audit[71586]: ANOM_ABEND auid=1000 uid=1000 gid=984 ses=3 pid=71586 comm="AudioRecv" exe="/usr/bin/moonlight" sig=11 res=1
Nov 02 20:57:09 nb-t480s kernel: AudioRecv[71670]: segfault at 10000000000 ip 00007f20f0d1184d sp 00007f203cff86d8 error 4 in libc.so.6[7f20f0be0000+15b000]
Nov 02 20:57:09 nb-t480s kernel: Code: 00 00 00 00 00 66 66 2e 0f 1f 84 00 00 00 00 00 66 66 2e 0f 1f 84 00 00 00 00 00 66 90 f3 0f 1e fa 48 89 f8 48 83 fa 20 72 23 <c5> fe 6f 06 48 83 fa 40 0f 87 a5 00 00 00 c5 fe 6f 4c 16 e0 c5 fe
```

Also the coredump in question:
[core.moonlight.1000.9bef0e4262634777a97aef6bc270c7d3.71586.1667419029000000.zst.tar.gz](https://github.com/moonlight-stream/moonlight-qt/files/9923924/core.moonlight.1000.9bef0e4262634777a97aef6bc270c7d3.71586.1667419029000000.zst.tar.gz)

**Comment by cgutman** [comment]
Fixed in v4.3.1

**Comment by transient-sepia** [comment]
Hi, I still have similar issues even on the latest version. Here's what changed compared to my last report:

```
Moonlight Version: 4.3.1
GeForce Experience version: 3.26.0.160
Nvidia GPU driver: 527.56
```

Excerpt from the stdout:

```
...
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1486)
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1487)
00:00:35 - SDL Info (0): Waiting for RFI frame
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1488)
00:00:35 - SDL Info (0): Waiting for RFI frame
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1489)
00:00:35 - SDL Info (0): Waiting for RFI frame
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1490)
00:00:35 - SDL Info (0): Waiting for RFI frame
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1491)
00:00:35 - SDL Info (0): Waiting for RFI frame
00:00:35 - SDL Info (0): Invalidate reference frame request sent (1486 to 1492)
00:00:35 - SDL Info (0): Next post-invalidation frame is: 1493 (P-frame)
[xcb] Unknown sequence number while processing queue
[xcb] You called XInitThreads, this is not your fault
[xcb] Aborting, sorry about that.
moonlight: xcb_io.c:278: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
Aborted (core dumped)
```

Coredump:
[core.moonlight.1000.1ca91f4aeafa4ff081033a3a7882a603.33815.1671241592000000.zst.tar.gz](https://github.com/moonlight-stream/moonlight-qt/files/10250539/core.moonlight.1000.1ca91f4aeafa4ff081033a3a7882a603.33815.1671241592000000.zst.tar.gz)
